python/rapidsmpf/rapidsmpf/integrations/dask/shuffler.py

```diff
@@ -147,9 +147,11 @@ class DaskIntegration(Protocol[DataFrameT]):
     @staticmethod
     def insert_partition(
         df: DataFrameT,
-        on: Sequence[str],
+        partition_id: int,
         partition_count: int,
         shuffler: Shuffler,
+        options: dict[str, Any],
```

Comment:
Sorry for the cross-posting, but https://github.com/rapidsai/cudf/pull/18720/files/94554a885f83021f72ce91ca04b96f0d67c6be28#diff-2d46acc0f845c678bb9acbbe8d6c14aaef1e85b63df306b87737a4f1cb3ece78 has a discussion about changing this type to `Any`, and then each implementation can (maybe) require a specific type like `TypedDict` with fields `on: Sequence[str]` for the implementation in cudf-polars.

Reply:
Thanks for the review @TomAugspurger - Do you think this PR is ready once https://github.com/rapidsai/cudf/pull/18720 is green?